src/Microsoft.Extensions.Localization/Internal/AssemblyResourceStreamManager.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.Extensions.Localization.Internal
+{
+    public class AssemblyResourceStreamManager : IResourceStreamManager
+    {
+        private const string _assemblyElementDelimiter = ", ";
+        private static readonly string[] _assemblyElementDelimiterArray = new[] { _assemblyElementDelimiter };
+        private static readonly char[] _assemblyEqualDelimiter = new[] { '=' };
+
+        private AssemblyWrapper _assembly;
+        private readonly string _resourceBaseName;
+
+        public AssemblyResourceStreamManager(
+            AssemblyWrapper resourceAssembly,
+            string resourceBaseName)
+        {
+            _assembly = resourceAssembly;
+            _resourceBaseName = resourceBaseName;
+        }
+        public string GetResourceStreamName(CultureInfo culture)
+        {
+            var resourceStreamName = _resourceBaseName;
+            if (!string.IsNullOrEmpty(culture.Name))
+            {
+                resourceStreamName += "." + culture.Name;
+            }
+            resourceStreamName += ".resources";
+
+            return resourceStreamName;
+        }
+
+        public string GetResourceStreamCacheKey(CultureInfo culture)
```

Comment:
But the public interface [here](https://github.com/aspnet/Localization/blob/dev/src/Microsoft.Extensions.Localization/IResourceNamesCache.cs#L20) defines it as a string so that would be a breaking API change. Unless I'm misunderstanding your ask?